server/lib/reverse_etl/utils/cursor_query_builder.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ReverseEtl
+  module Utils
+    class CursorQueryBuilder
+      def self.update_model_query(sync_config)
+        existing_query = sync_config.model.query
+
+        if sync_config.cursor_field && sync_config.current_cursor_field
+          new_query = "(#{existing_query}) AS subquery " \
+          "WHERE #{sync_config.cursor_field} >= '#{sync_config.current_cursor_field}' " \
+          "ORDER BY #{sync_config.cursor_field} ASC"
+        elsif sync_config.cursor_field
+          new_query = "(#{existing_query}) AS subquery " \
```

Comment:
Why do we need as subquery? Can't we just use existing_query?

Reply:
In some cases, additional conditions directly to the existing query without using a subquery could lead to ambiguity or unexpected behavior, 
 if the original query already contains conditions or clauses. Wrapping it in a subquery helps to isolate and clarify the scope of each condition.